### ISSUE TITLE
Design tooltip component

### ISSUE THREAD

**Issue by kimisgold** [omeka/Omeka#1061]
We've been relying on the `title` attribute to provide a tooltip, especially where we have icons without text labels. This is problematic for a number of reasons, including inaccessibility to users on touch devices as well as repetitive label readouts on screen readers. Here we'll gather examples and discuss requirements for what we want from a tooltip.

**Comment by aviannamiller** [comment]
Also note the tooltip pattern seems to be applying to site searches, at least on some themes (Freedom and Lively, for sure, potentially also themes like Seasons that offer "search options"):
<img width="664" height="133" alt="Search form on site with Lively theme. Within the submit button, there is a label inserted that reads '? Submit'." src="https://github.com/user-attachments/assets/d074a267-9c23-4e59-ad8e-ccd71e3278b2" />
<img width="663" height="152" alt="Same search form, though this time the word 'Search' appears under the search submit icon." src="https://github.com/user-attachments/assets/d0a2c5ac-55bf-41b2-8ea6-ac198060116b" />

**Comment by kimisgold** [comment]
I've done a JS experiment positioning tooltips for older browsers, repositioning the tooltip if its default position doesn't clear the height of its parent element. This should only impact the browsers that have the tooltips overlaying their parent buttons.

I've also fixed the dialog sizing on smaller device widths, so hopefully that should help with some of the awkwardness you're seeing.

**Comment by aviannamiller** [comment]
Overall looks good on Safari on mobile, Chrome on mobile, Mac, and Windows, Edge, Firefox on Mac **and** Windows! Spacing should be adjusted a bit to fully meet the standards of the content on hover / focus SC, see below for more details.

On Safari on desktop, there is some slightly odd positioning happening with narrow viewports / the mobile layout (but things do look fine on mobile), among other things:

https://github.com/user-attachments/assets/672087da-8441-44d2-8196-6e3d66215dad

Video description: Keyboarding through search form buttons to reveal tooltips on Classic admin dashboard. With focus, tooltips appear centered below buttons, with quite a bit of distance in between. Activating one of the buttons opens a modal with a close button that also has a tooltip, but that tooltip is situated below but slightly to the right of the button.

Note that focus outlines are also grabbing onto the tooltips. This is leaving the indicator around the tooltip to linger at times, especially in the example of the close button in the modal (see above). I am exclusively noticing this on Safari, perhaps more clearly modeled by this screenshot with a VO focus indicator layered over:
<img width="401" height="118" alt="Detail of search options button. Focus outline wraps around both the button and its associated tooltip. A larger focus indicator encases both components." src="https://github.com/user-attachments/assets/6367ec05-4c51-44a3-ba33-03fd776c2b40" />
I am also finding that tooltips are not properly dismissing on Safari. They dismiss with an esc key press on Firefox and Chrome.

The spacing issue is creating a problem with not being able to hover over the tooltips. This kind of content must be hoverable (in addition to persistent). Technically, this is a problem with the positioning on **all** tested browsers. The space between the button and tooltip probably needs to be tightened up enough across the board so one could move their cursor hover directly from the button to the tooltip with it still persisting.

**Comment by kimisgold** [comment]
Awesome, this is good progress. While I figure out these quirks with the focus outline and close on Safari, I'm going to apply the pattern across the admin in this branch so we can start testing out other use cases.

**Comment by kimisgold** [comment]
The tooltips should now appear wherever we rely on fonticon glyphs in the interface. Hitting "escape" in Safari should now dismiss them. (Apparently `popover="hint"` isn't yet fully supported in Safari. I think `popover="auto"` pretty much does what we want it to, and `hint` is mainly used for not dismissing `auto` popovers.)

I still need to look into the focus outline quirk, but everything else is ready for testing.

**Comment by aviannamiller** [comment]
For the most part, tooltips are working great in terms of being hoverable, persistent, and dismissible.

- There are some positioning issues on Chrome and Edge on Windows and Chrome and Safari on Mac, definitely more overt for the latter. These are apparent when the viewport is resized, mostly when zooming in or narrowing the browser width. The following screen recording is from Safari, but Chrome (Mac) has something very similar going on with the expand/collapse toggles and move icons.
  
  https://github.com/user-attachments/assets/b31eea47-a061-4f67-9bf8-a4f14dae3aeb

  Video description: Admin interface. Mousing over pagination arrows on browse view prompts “Previous Page” and “Next Page” tooltips, which are situated below but slightly to the right of pagination elements. Tab is switched to navigation configuration page. Hovering over expand/collapse toggles prompts a matching tooltip below a given toggle, but as the page is zoomed into, the tooltip changes positions, floating quite a distance above the toggle.
  
  For the aforementioned browsers on Windows, it’s a bit more subtle; there’s more of an issue of the tooltip at times appearing to overlay the toggle instead of positioning directly below.
  
  Also for Safari, some odd positioning with elements that are located close to the edge of their containers, like in the above example with the pagination arrows. That seems to be exclusively a Safari issue (or at least a my-version-of-Safari issue), in addition to the following:
  <img width="649" height="197" alt="'Move' tooltip, associated with move icon. Located below but slightly to the right of the icon." src="https://github.com/user-attachments/assets/29f38b84-8a7e-4c13-a284-dfa3e48d14c9" />
  <img width="616" height="128" alt="'Admin' tooltip, located below but slightly to the right of three dots and cut off on edge of screen." src="https://github.com/user-attachments/assets/ad3f2235-6aa1-44d3-9576-04c416967f82" />
- Items marked for deletion with undo icons still have tooltips that read “Remove”.
- Tooltips appear to be missing from: 
    - #content-nav-toggle for mobile menu.
    - Pagination, ascending/descending icons, and delete buttons on Tags.
    - Delete button on Delete Item Type dialog.
    - Expand/collapse and delete/undo buttons on newly added nav items when editing Navigation.
    - Sort icons on browse views? Would tooltips make sense here to indicate sort order?

**Comment by kimisgold** [comment]
Still poking at some of your other comments, but I just put in a fix for horizontal positioning. Can you check to see if the tooltips are not horizontally centered in any of the problem browsers?

**Comment by aviannamiller** [comment]
They now look to be horizontally centered across all browsers.

**Comment by kimisgold** [comment]
A couple things: I've just put in a fix to account for window resizing when the user zooms. I've also experimented with removing the polyfill. It seems to work on Firefox 142.0.1 on Windows, where I've been replicating the biggest issues. Definitely curious to see if it's the same for Safari 17.6.

**Comment by aviannamiller** [comment]
I'm not sure that there's an improvement on my end. I'm now seeing the issue with the tooltip and the distance from the trigger element when zooming on Chrome (148.0.7778.168), Edge (148.0.3967.70), and Firefox (must have updated: 149.0.2) on Windows and Chrome (148.0.7778.98), Safari, and Firefox (151.0) on Mac. Previously, Firefox had seemed relatively fine in terms of this specific problem.

Perhaps this wasn't ready to be reviewed at this point anyway, but I noticed the tooltip text on a nav item marked for deletion also turns red with the rest of the item.

**Comment by kimisgold** [comment]
Okay, tooltip should always be relying on the JS calculating its position now. It seems to be positioning consistently across my macOS browsers, even on resizing and zooming.

**Comment by aviannamiller** [comment]
Unfortunately, the issues persist on my end. To confirm, I am on `accessible-tooltip-popover`. The maybe good news is that the issues look to be the same across all browsers. For your sake, my up-to-date browser versions are as follows: Chrome v.152.0.7977.75, Safari v.26.5, Firefox v.155.0.

Here's a screen recording from Chrome:

https://github.com/user-attachments/assets/78a0b897-3f38-43d3-91de-ec519b868048

Video description: Scrolling through admin-side Navigation Appearance page. While scrolling, the cursor hovers over the expand/collapse toggle on navigation item drawers, exposing Expand tooltips, clicking on some to show Collapse tooltips. As the page is scrolled down, vertical distance appears between the location of the expand/collapse toggles and the tooltips.

And from Firefox, focus is noted by the location of the red circle:

<img width="823" height="719" alt="Detail of navigation link drawers in appearance tab. Red circle outlines a drag handle on one of the drawers, but its associated tooltip with a label of 'Move' is floating above, about the combined height of two drawer containers up." src="https://github.com/user-attachments/assets/f61092e7-697c-45b8-be85-2596627888dc" />

It seems like the further you scroll, the more distance that grows between the trigger element and the tooltip. This is regardless of browser zoom level as well as if operating via pointer or keyboard.
